Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/Game.java

```diff
@@ -0,0 +1,115 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.util.ArrayList;
+
+public class Game {
+	protected 	Dictionary 	dictionary;		
+	protected 	Player 		player;		  	
+	protected 	char[] 		etalon;			
+	protected	ArrayList 	<Character> 	guessedLetters;
+	protected	char[] 		openedWord;		
+	protected	ArrayList	<Character>		wrongLetters;
+	protected 	UserInterface ui;			
+	protected 	final int 	TOTAL_GUESSES=9;
+	protected 	int 		currentGuesses;
+
+
+	public Game(){
+		dictionary = new ArrayDictionary ("words.txt");
+		ui = new GUIMainWindow(this);
```

Comment:
Not sure why the Game logic knows anything about the UI, the rules of the game do not depend on how it is displayed at all. The UI is a way to manipulate the game. Hence, the UI relies on the game and the game exists regardless of whether there is a way to display it. Therefore, the Game should hold a reference to Game and manipulate that. Game should know nothing about UI, and instead the UI should pull state from the Game and display it to the user. So.... I think this is a code smell that you've not separated responsibility between classes nicely.